btk/survey.py

```diff
@@ -0,0 +1,243 @@
+import astropy.wcs as WCS
+import numpy as np
+import galsim
+from collections import namedtuple
+
+# A simple class archetype to serve as a dictionary without having to write the field names
+# every time.
+Survey = namedtuple(
+    "Survey",
+    [
+        "name",
+        "pixel_scale",
+        "effective_area",
+        "mirror_diameter",
+        "airmass",
+        "filters",
+    ],
+)
+
+Filter = namedtuple(
+    "Filter",
+    [
+        "name",
+        "psf_scale",
+        "sky_brightness",
+        "exp_time",
+        "zenith_psf_fwhm",
+        "zero_point",
+        "extinction",
+    ],
+)
+
+
+pix_roman = 0.11
+pix_rubin = 0.2
+pix_hst = 0.06
+pix_euclid = 0.101
+pix_hsc = 0.167
+pix_des = 0.263
+pix_cfht = 0.185
+
+# Central wavelengths in Angstroms for each LSST filter band, calculated from the
+# baseline total filter throughputs tabulated at
+# http://dev.lsstcorp.org/cgit/LSST/sims/throughputs.git/snapshot/throughputs-1.2.tar.gz
+_central_wavelength = {
+    "u": 3592.13,
+    "g": 4789.98,
+    "r": 6199.52,
+    "i": 7528.51,
+    "z": 8689.83,
+    "y": 9674.05,
+    "VIS": 7135.0,
+}
+
+
+# https://sci.esa.int/documents/33859/36320/1567253682555-Euclid_presentation_Paris_1Dec2009.pdf
+Euclid = Survey("Euclid", pix_euclid, [Filter("VIS", 0.16, 22.9, 2260, 6.85)])
+
+# Source https://hst-docs.stsci.edu/display/WFC3IHB/6.6+UVIS+Optical+Performance#id-6
+# .6UVISOpticalPerformance-6.6.1 800nm
+HST = Survey("HST", pix_hst, [Filter("f814w", 0.074, 22, 3000, 20)])
+
+# https://hsc-release.mtk.nao.ac.jp/doc/ deep+udeep
+
+HSC = Survey(
+    "HSC",
+    pix_hsc,
+    [
+        Filter("g", 0.306, 21.4, 600, 91.11),
+        Filter("r", 0.285, 20.6, 600, 87.74),
+        Filter("i", 0.238, 19.7, 1200, 69.80),
+        Filter("z", 0.268, 18.3, 1200, 29.56),
+        Filter("y", 0.272, 17.9, 1200, 21.53),
+    ],
+)
+
+# Sigma of the psf profile in arcseconds.
+# https://arxiv.org/pdf/1702.01747.pdf Z-band
+Roman = Survey(
+    "Roman",
+    pix_roman,
+    [
+        Filter("F062", 0.1848, 22, 3000, 26.99),
+        Filter("Z087", 0.1859, 22, 3000, 26.39),
+        Filter("Y106", 0.2046, 22, 3000, 26.41),
+        Filter("J129", 0.2332, 22, 3000, 26.35),
+        Filter("H158", 0.2684, 22, 3000, 26.41),
+        Filter("F184", 0.2981, 22, 3000, 25.96),
+    ],  # Mean sky level and exposure time need to be checked
+)
+
+# https://www.lsst.org/about/camera/features
+Rubin = Survey(
+    "LSST",
+    pix_rubin,
+    32.4,
+    8.36,
+    1.2,
+    [
+        Filter("y", 0.327, 18.6, 4800, 0.63, 10.58, 0.138),
```

Comment:
Are the zenith_psf_fwhm in the right order ? They go upward while psf_scale is going downward.

Reply:
corrected thanks to @aboucaud comment of using keywords.